### ISSUE TITLE
Use GHC call stacks

### ISSUE THREAD

**Issue by skogsbaer** [skogsbaer/HTF#72]
Macros are no longer required to get the location of an assertion failure. Also, the subAssert combinator is no longer necessary.

But macros are still needed for collecting test cases.

**Comment by skogsbaer** [comment]
Fixed in #115 and version 0.15.0.0

**Comment by andreasabel** [comment]
@skogsbaer: Is there a way to clear the error location?  
This is how `bnfc-system-tests` wrapped `assertEqual_` etc.: https://github.com/BNFC/bnfc/blob/69e2b3b026e993ff87012f3efb73fd441714185d/testing/src/TestUtils.hs#L47-L49
With `HTF-0.15`, I have to migrate this to `assertEqual`.  This is easy enough, but I get now an uninteresting error location in my test output which I would like to suppress:
```
[TEST] Testing BNFC on ill-formed LBNF input:fail-lbnf/duplicate-cons
bnfc --check fail-lbnf/duplicate-cons.cf
assertEqual failed at src/TestUtils.hs:48
* expected: 0
* but got:  1
* diff:     01
  at src/TestUtils.hs:48
*** Failed! (19ms)
```
The info `at src/TestUtils.hs:48` isn't helpful in my case since it does not point me to the source of the problem.
